lib/govuk_index/page_traffic_loader.rb

```diff
@@ -0,0 +1,48 @@
+module GovukIndex
+  class PageTrafficLoader
+    def initialize(iostream_batch_size: 250)
+      @iostream_batch_size = iostream_batch_size
+      @logger = Logging.logger[self]
+      @logger.level = :info
+    end
+
+    def load_from(iostream)
+      new_index = index_group.create_index
+      @logger.info "Created index #{new_index.real_name}"
+
+      old_index = index_group.current_real
+      @logger.info "Old index #{old_index.real_name}"
+
+      old_index.with_lock do
+        @logger.info "Indexing to #{new_index.real_name}"
+
+        in_even_sized_batches(iostream) do |lines|
+          GovukIndex::GooglePopularityWorker.perform_async(lines, new_index.real_name)
+        end
+
+        GovukIndex::GooglePopularityWorker.wait_until_processed
+        new_index.commit
+
+        # Switch aliases inside the lock so we avoid a race condition where a
+        # new index exists, but the old index is available for writes
+        index_group.switch_to(new_index)
+        old_index.close
+      end
+    end
+
+  private
+
+    # Breaks the given input stream into batches of documents
+    def in_even_sized_batches(iostream, batch_size = @iostream_batch_size, &_block)
```

Comment:
Do you think we should add a link to the [bulk API docs](https://www.elastic.co/guide/en/elasticsearch/reference/2.4/docs-bulk.html) to explain why things need to be split into even-sized batches? It's pretty counterintuitive if you haven't used the bulk API before.